make deprecation message less confusing

WebFreak001 opened the pull request:
the deprecation message talks about the function we are using, not some arbitrary example that you would use "such as" with.

thewilsonator reviewed (approved):
is this message not tested anywhere?

WebFreak001 commented:
just haven't tested it locally, letting CI run through this since I edited it on github ^^

maxhaton commented on compiler/src/dmd/traits.d:
```diff
@@ -1225,7 +1225,7 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
                 if (td.overnext || td.funcroot)
                 {
-                    deprecation(e.loc, "`__traits(getAttributes)` may only be used for individual functions, not overload sets such as: `%s`", td.ident.toChars());
+                    deprecation(e.loc, "`__traits(getAttributes)` may only be used for individual functions, not overload set `%s`", td.ident.toChars());
```
```suggestion
                    deprecation(e.loc, "`__traits(getAttributes)` may only be used for individual functions, not the overload set `%s`", td.ident.toChars());
```
Also DRY

WebFreak001 committed:
make deprecation message less confusing

the deprecation message talks about the function we are using, not some arbitrary example that you would use "such as" with.
